opentaxii/taxii2/validation.py

```diff
@@ -0,0 +1,29 @@
+import json
+
+from opentaxii.taxii2.exceptions import ValidationError
+from stix2 import parse
+from stix2.exceptions import STIXError
+
+
+def validate_envelope(json_data: str, allow_custom: bool = False) -> None:
+    """
+    Validate if ``json_data`` is a valid taxii2 envelope.
+
+    :param str json_data: the data to check
+    :param bool allow_custom: if true, allow non-standard stix types
+    """
+    if not json_data:
+        raise ValidationError("No data.")
+    try:
+        data = json.loads(json_data)
+    except json.JSONDecodeError as e:
+        raise ValidationError(f"Invalid json. {str(e)}.") from e
+    if "objects" not in data:
+        raise ValidationError("No objects.")
+    for item in data["objects"]:
+        try:
+            parse(item, allow_custom)
```

Comment:
You can simplify it by tring to make a `Bundle(objects=data['objects'], allow_custom=allow_custom)`.

Reply:
Thanks for the suggestion. I tried it out, but it results in a less helpful error message:

```
Invalid value for Bundle 'objects': No values for required properties for Indicator: (labels, pattern).
```

This doesn't include information on *where* the problem is. The current implementation does, so I'm keeping it.